tfsdk/request_validation.go

```diff
@@ -0,0 +1,46 @@
+package tfsdk
+
+// ValidateDataSourceConfigRequest represents a request to validate the
+// configuration of a data source. An instance of this request struct is
+// supplied as an argument to the DataSource ValidateConfig receiver method
+// or automatically passed through to each ConfigValidator.
+type ValidateDataSourceConfigRequest struct {
+	// Config is the configuration the user supplied for the data source.
+	//
+	// This configuration may contain unknown values if a user uses
+	// interpolation or other functionality that would prevent Terraform
+	// from knowing the value at request time.
+	Config Config
+
+	// TypeName is the data source type name.
+	TypeName string
```

Comment:
Do we want to surface this? I think we opted not to in other places....

Reply:
I'm guessing it could be useful in the case where there are multiple resources against the same remote system that are similar, but slightly different. If we're not passing through this information that's available over the wire though in other places, happy to remove this for now (we can always add it back).

Reply:
Removed in 572ee67c06437fed09622e9a74e7b046422acdeb